packages/functional-tests/tests/subscription-tests/coupons-ui-tests/coupon.spec.ts

```diff
@@ -30,6 +26,12 @@ test.describe('severity-2 #smoke', () => {
       // 'autoexpired' coupon is an expired coupon for a 6mo plan
       await subscribe.addCouponCode('autoexpired');
 
+      await expect(
+        await subscribe.getCouponStatusByDataTestId('coupon-error')
+      ).toBeVisible({
+        timeout: 5000,
+      });
+
       // Verifying the correct error message
       expect(await subscribe.couponErrorMessageText()).toContain(
         'The code you entered has expired'
```

Comment:
I'm wondering if we would get more reliable performance if we made this an auto-awaited assert that is user-facing.

```
await expect(page.getByText('The code you entered has expired')).toBeVisible();
```